common/src/main/java/org/opensearch/security/common/package-info.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/**
+ * This package defines common classes required to implement resource access control in OpenSearch.
```

Comment:
It is unclear to me why these classes need to be moved into an own module. I see that the `client` module depends on it via an `implementation` dependency, right? So, will each plugin then carry a JAR with these implementations?

As some of them feel rather security-core-ish, like `AdminDNs`, I am not sure if it is good to move these out of the security core.